accelerator/common_core.c

```diff
@@ -402,6 +400,12 @@ int ta_find_transactions_obj_by_tag(const iota_client_service_t* const service,
   // get transaction obj
   for (hash_trits = hash243_queue_peek(hash_res->hashes); hash_trits != NULL;
        hash_trits = hash243_queue_peek(hash_res->hashes)) {
+    ta_get_transaction_object_res_t* obj_res =
+        ta_get_transaction_object_res_new();
```

Comment:
Why do you move this in loop?

Reply:
Explained in the commit message. Basically, if there are multiple transactions, only the last one will be freed by `ta_get_transaction_object_res_free`.